### PR TITLE
show mesh layer icon in layer tree #20722

### PR DESCRIPTION
**Pull request by PeterPetrik**
fix #20722

### CODE REVIEW

File: src/core/layertree/qgslayertreemodel.cpp [hunk 1]
```diff
@@ -199,6 +199,10 @@ QVariant QgsLayerTreeModel::data( const QModelIndex &index, int role ) const
       {
         return QgsLayerItem::iconRaster();
       }
+      else if ( layer->type() == QgsMapLayer::MeshLayer )
```
Comment (nyalldawson) [inline src/core/layertree/qgslayertreemodel.cpp]: Could you change this block to a switch so we get warnings if new layer types are added in future?

Comment (PeterPetrik) [reply src/core/layertree/qgslayertreemodel.cpp]: sure I can, but I am not sure I have time to add another layer type this year... I say, I layer type per decade is fine with me.

Comment (PeterPetrik) [reply src/core/layertree/qgslayertreemodel.cpp]: @nyalldawson looking at the code the logic there is quite complex and not sure if I can refactor it to switch without change of behaviour for plugin layers. Could we postpone the refactoring after 3.6?

**Comment by rduivenvoorde** [comment]
@PeterPetrik nice, thanks! 

mesh

**Pull request merged**